### ISSUE TITLE
Unicode support for filenames

### ISSUE THREAD

**Issue by srossross** [ContinuumIO/odo#262]
Odo fails when filename is unicode string:

```
In [11]: odo(u'/Users/sean/Library/Application Support/binstar/conda-data/sean+sample-data+2015.07.08+1118+us-500.csv', pd.DataFrame)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-ac4485c79f7c> in <module>()
----> 1 odo(u'/Users/sean/Library/Application Support/binstar/conda-data/sean+sample-data+2015.07.08+1118+us-500.csv', pd.DataFrame)

/Users/sean/anaconda/lib/python2.7/site-packages/odo/odo.pyc in odo(source, target, **kwargs)
     88     odo.append.append      - Add things onto existing things
     89     """
---> 90     return into(target, source, **kwargs)

/Users/sean/anaconda/lib/python2.7/site-packages/multipledispatch/dispatcher.pyc in __call__(self, *args, **kwargs)
    161             self._cache[types] = func
    162         try:
--> 163             return func(*args, **kwargs)
    164 
    165         except MDNotImplementedError:

/Users/sean/anaconda/lib/python2.7/site-packages/odo/into.pyc in into_type(a, b, **kwargs)
     23         if 'dshape' not in kwargs:
     24             kwargs['dshape'] = discover(b)
---> 25     return convert(a, b, **kwargs)
     26 
     27 

/Users/sean/anaconda/lib/python2.7/site-packages/odo/core.pyc in __call__(self, *args, **kwargs)
     28 
     29     def __call__(self, *args, **kwargs):
---> 30         return _transform(self.graph, *args, **kwargs)
     31 
     32 

/Users/sean/anaconda/lib/python2.7/site-packages/odo/core.pyc in _transform(graph, target, source, excluded_edges, ooc_types, **kwargs)
     41     pth = path(graph, type(source), target,
     42                excluded_edges=excluded_edges,
---> 43                ooc_types=ooc_types)
     44     try:
     45         for (A, B, f) in pth:

/Users/sean/anaconda/lib/python2.7/site-packages/odo/core.pyc in path(graph, source, target, excluded_edges, ooc_types)
     77                                     if issubclass(n, oocs)])
     78     with without_edges(graph, excluded_edges) as g:
---> 79         pth = nx.shortest_path(g, source=source, target=target, weight='cost')
     80         result = [(src, tgt, graph.edge[src][tgt]['func'])
     81                   for src, tgt in zip(pth, pth[1:])]

/Users/sean/anaconda/lib/python2.7/site-packages/networkx/algorithms/shortest_paths/generic.pyc in shortest_path(G, source, target, weight)
    136                 paths=nx.bidirectional_shortest_path(G,source,target)
    137             else:
--> 138                 paths=nx.dijkstra_path(G,source,target,weight)
    139 
    140     return paths

/Users/sean/anaconda/lib/python2.7/site-packages/networkx/algorithms/shortest_paths/weighted.pyc in dijkstra_path(G, source, target, weight)
     75     """
     76     (length, path) = single_source_dijkstra(G, source, target=target,
---> 77                                             weight=weight)
     78     try:
     79         return path[target]

/Users/sean/anaconda/lib/python2.7/site-packages/networkx/algorithms/shortest_paths/weighted.pyc in single_source_dijkstra(G, source, target, cutoff, weight)
    337                 edata.append((w, {weight: minweight}))
    338         else:
--> 339             edata = iter(G[v].items())
    340 
    341         for w, edgedata in edata:

/Users/sean/anaconda/lib/python2.7/site-packages/networkx/classes/graph.pyc in __getitem__(self, n)
    317         {1: {}}
    318         """
--> 319         return self.adj[n]
    320 
    321 

KeyError: <type 'unicode'>

```

**Comment by cpcloud** [comment]
you're on an older version of odo

```
In [20]: df = pd.DataFrame({'a': [1,2,3], 'b':[4,5,6],'c':[7,8,9],'d':[10,11,12]})

In [21]: df.to_csv('foo.csv', index=False, sep=',')

In [22]: odo(u'foo.csv', pd.DataFrame, sep=',')
Out[22]:
   a  b  c   d
0  1  4  7  10
1  2  5  8  11
2  3  6  9  12
```

**Comment by cpcloud** [comment]
I just released a new version 0.3.3, which has this fix in it
